### PR TITLE
Fix callback docstrings

### LINKED ISSUE #20965: Improve Mlflow Callbacks documentation.
I recently followed https://julsimon.medium.com/using-mlflow-with-hugging-face-transformers-4f69093a6c04 , transformers documentation on mlflow callback is not formatted properly.

image

It is better to read from the source code https://github.com/huggingface/transformers/blob/accad48e5b4a98302ea396b9f15c5f1c987b6f7f/src/transformers/integrations.py#L894 than the documentation site. 
image

I am sure that in the past, I might have seen such examples over other classes. I thought may be I  should report it.

### PR DESCRIPTION
**Pull request by stevhliu**
Fixes #20965 where the parameters aren't properly formatted because it uses `Environment` instead of `Args` in the docstring.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
This is not the proper fix as those constants are not arguments of the function but environment variables. The list needs to be formatted in Markdown syntax instead.

**Comment by stevhliu** [comment]
Reformatted as Markdown 👍

File: src/transformers/integrations.py [hunk 1]
```diff
@@ -668,16 +668,16 @@ def setup(self, args, state, model, **kwargs):
...
-                Whether or not to log model as artifact at the end of training. Use along with
-                *TrainingArguments.load_best_model_at_end* to upload best model.
-            WANDB_WATCH (`str`, *optional* defaults to `"gradients"`):
-                Can be `"gradients"`, `"all"` or `"false"`. Set to `"false"` to disable gradient logging or `"all"` to
-                log gradients and parameters.
-            WANDB_PROJECT (`str`, *optional*, defaults to `"huggingface"`):
-                Set this to a custom string to store results in a different project.
-            WANDB_DISABLED (`bool`, *optional*, defaults to `False`):
-                Whether or not to disable wandb entirely. Set *WANDB_DISABLED=true* to disable.
+        - WANDB_LOG_MODEL (`bool`, *optional*, defaults to `False`):
```
Comment (sgugger) [inline src/transformers/integrations.py:671]:
```suggestion
        - **WANDB_LOG_MODEL** (`bool`, *optional*, defaults to `False`):
```
If we want to have the formatting as the arguments of a function/class, we can put all the names in bold.

**Review by sgugger — APPROVED** [review]
Thanks for iterating! Just left a suggestion (to apply on all env variable names if you like it).

**Pull request merged**